dtls_debug.h

```diff
@@ -78,11 +78,11 @@ void dtls_set_log_level(log_t level);
  * level is below or equal to the log level that set by
  * set_log_level(). */
 #ifdef HAVE_VPRINTF
-#if (defined(__GNUC__))
+#if (defined(__GNUC__) && !defined(__MINGW32__))
```

Comment:
@mrdeep1 Could it be that (some of) the macOS related warnings also result from here? If I am not mistaken, clang also defines `__GNUC__`, so maybe we need to add `&& !defined(__clang__)` or `!defined(__llvm__)` here to get rid of the warnings.

Reply:
I tried out adding `&& !defined(__clang__)` but it did not have any effect on the macOS warnings. I suppose this should be addressed in a follow-up PR then.

Reply:
AFAIK, these MAC OS warnings are not related to the changes here in this PR.
These warnings occurs also on other PRs without the changes of this one.
So, no need to care.